### PR TITLE
add e2e for migration rollback feature

### PR DESCRIPTION
**Pull request by wulemao**
**What type of PR is this?**

/kind failing-test

**What this PR does / why we need it**:

add e2e for migration rollback feature

**Which issue(s) this PR fixes**:

Fixes part of #5577

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by a7i** [comment]
@wulemao  thanks for your patience, would you please rebase since the PR was merged?

File: pkg/dependenciesdistributor/dependencies_distributor.go [hunk 1]
```diff
@@ -546,6 +546,7 @@ func (d *DependenciesDistributor) removeScheduleResultFromAttachedBindings(bindi
 		delete(attachedBindings[index].Labels, bindingLabelKey)
 		updatedSnapshot := deleteBindingFromSnapshot(bindingNamespace, bindingName, attachedBindings[index].Spec.RequiredBy)
 		attachedBindings[index].Spec.RequiredBy = updatedSnapshot
+		attachedBindings[index].Spec.ConflictResolution = ""
```
Comment (XiShanYongYe-Chang) [inline pkg/dependenciesdistributor/dependencies_distributor.go:549]: Do you need the latest code for rebase?

Comment (wulemao) [reply pkg/dependenciesdistributor/dependencies_distributor.go:549]: already rebased your PR, waiting for #5731 or #4418 to be merged and do rebase

Comment (wulemao) [reply pkg/dependenciesdistributor/dependencies_distributor.go]: done

**Comment by wulemao** [comment]
done, any more comments?

**Comment by XiShanYongYe-Chang** [comment]
Thanks
/approve

File: test/e2e/migration_and_rollback_test.go [hunk 2]
```diff
@@ -30,17 +30,30 @@ import (
...
 	"github.com/karmada-io/karmada/pkg/util/names"
 	"github.com/karmada-io/karmada/test/e2e/framework"
 	"github.com/karmada-io/karmada/test/helper"
 )
 
-var _ = ginkgo.Describe("Seamless migration testing", func() {
+var _ = ginkgo.Describe("Seamless migration and rollback testing", func() {
 	var member1 string
 	var member1Client kubernetes.Interface
+	karmadaLabels := []string{workv1alpha2.ResourceBindingPermanentIDLabel, workv1alpha2.WorkPermanentIDLabel, util.ManagedByKarmadaLabel}
```
Comment (a7i) [inline test/e2e/migration_and_rollback_test.go:46]: it doesn't seem right to implement this in tests. Ideally we build a var that both the application logic and test logic use

Comment (a7i) [reply test/e2e/migration_and_rollback_test.go:46]: I can submit a PR to demonstrate

Comment (a7i) [reply test/e2e/migration_and_rollback_test.go:46]: https://github.com/karmada-io/karmada/pull/5747

**Pull request merged**